### PR TITLE
Implement HTMLCollection @@iterator

### LINKED ISSUE #1600: Iterator regression in 9.5
There is a regression in 9.5 compared to 9.4.2 (and before — this is code that has worked for roughly a year). The following used to work:

```js
for (let $section of document.getElementsByTagName('section')) // …
```

Now it fails with the classic `TypeError: document.getElementsByTagName(...)[Symbol.iterator] is not a function`. I believe the change that introduced the problem was:

https://github.com/tmpvar/jsdom/commit/4b80cd674418736fe4cde25414699440ece07d4c

### PR DESCRIPTION
**Pull request by i8-pi**
Fixes #1600

### CODE REVIEW

**Comment by i8-pi** [comment]
Not sure if fixing eslint warnings is correct since upstream doesn't seem to use the same style or any ES6 features. If it is correct, I can squash the commits

File: test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!doctype html>
+<meta charset="utf-8">
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html]: Not required, but it's always nice to add a few link rel=help to the spec links we discussed earlier, and you can add a link rel=author to get some form of lasting credit :)

File: test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<title>HTMLCollection @@iterator Test</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+    <p id="1"></p>
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html]: Let's not indent these.

File: test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <p id="2"></p>
+    <p id="3"></p>
+    <p id="4"></p>
+    <p id="5"></p>
+<script>
+  "use strict";
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html]: No indent in the script

File: test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  let paragraphs;
+  setup(() => {
+    paragraphs = document.querySelectorAll("p");
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html]: No need for setup here. Just use `const paragraphs = ...` at the top of the script.

File: test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  test(() => {
+    assert_false("values" in paragraphs);
+  }, "HtmlCollection does not have iterable's values method.");
+  test(() => {
+    assert_false("entries" in paragraphs);
+  }, "HtmlCollection does not have iterable's entries method.");
+  test(() => {
+    assert_false("forEach" in paragraphs);
+  }, "HtmlCollection does not have iterable's forEach method.");
+  test(() => {
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/dom/collections/HTMLCollection-iterator.html]: A newline between each test would be lovely.

**Comment by domenic** [comment]
Looks great! Minor test nits that would be nice to fix, but nothing that will stop this from landing; if you're too busy just let me know.

**Comment by i8-pi** [comment]
All fixed and ready for round 2

**Pull request merged**